cmake/modules/AddSwift.cmake

```diff
@@ -127,17 +127,17 @@ function(_add_variant_c_compile_link_flags)
   set(_sysroot "${SWIFT_SDK_${CFLAGS_SDK}_ARCH_${CFLAGS_ARCH}_PATH}")
   if(IS_DARWIN)
     list(APPEND result "-isysroot" "${_sysroot}")
+  elseif("${CFLAGS_SDK}" STREQUAL "ANDROID")
+    list(APPEND result "--sysroot=${_sysroot}")
```

Comment:
If the values really are the same, why not just leave this as a separate `if` but drop the `--sysroot` flag?

Reply:
Yeah, they are the same now, especially now that I've split out the linker handling.  Are you suggesting this instead:

```
if(IS_DARWIN)
   ...
elseif(NOT SWIFT_COMPILER_IS_MSVC_LINK AND NOT "${_sysroot}" STREQUAL "/")
  list(APPEND result "--sysroot=${_sysroot}")
  if("${CFLAGS_SDK}" STREQUAL "ANDROID")
    if(NOT SWIFT_ENABLE_LLD_LINKER)
      ...
    endif()
  endif()
endif()
```

Reply:
I don't think the Android part belongs inside this condition at all. This `if` is handling the sysroot; setting the linker isn't related to that.

Reply:
@jrose-apple oh, I see, you want to move the linker handling out of it?  Okay.  It was kinda messy before, but this is a different type of messy.  I see your point.  Want to cancel this or should I just do that as a follow up?